### PR TITLE
Fix memory leak for images resize command.

### PR DESCRIPTION
**Pull request by Koc**
### Description
We have large catalog with many products and 2Gb memory limit is not enough to resize all of the images. I've removed storing loaded products inside identity map and it helps a lot.

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by Koc** [comment]
I can also add progressbar to indicate resizing progress if this acceptable

**Comment by mslabko** [comment]
Hi @Koc ,
For mainline this issue already fixed - https://github.com/magento/magento2/blob/develop/app/code/Magento/Catalog/Model/ProductRepository.php#L311-L312 
I think it make sense to port existing solution to 2.1, does it work for you?

**Comment by Koc** [comment]
@mslabko I haven't tested this yet but looks like it should works. Agree, it is better backport it to 2.1. But who  should do it: am I in this PR or somewho from Magento's team can?

**Comment by ishakhsuvarov** [comment]
@Koc I am pretty sure you can work on this, if you wish

File: app/code/Magento/Catalog/Api/ProductRepositoryInterface.php [hunk 1]
```diff
@@ -72,4 +72,11 @@ public function deleteById($sku);
      * @return \Magento\Catalog\Api\Data\ProductSearchResultsInterface
      */
     public function getList(\Magento\Framework\Api\SearchCriteriaInterface $searchCriteria);
+
+    /**
+     * Clean internal product cache
+     *
+     * @return void
+     */
+    public function cleanCache();
```
Comment (okorshenko) [inline app/code/Magento/Catalog/Api/ProductRepositoryInterface.php:81]: it is not allowed change @api interfaces in patch releases. Unfortunately, we can not accept this change.

File: app/code/Magento/Catalog/Console/Command/ImagesResizeCommand.php [hunk 2]
```diff
@@ -95,15 +94,17 @@ protected function execute(InputInterface $input, OutputInterface $output)
                 $imageCache = $this->imageCacheFactory->create();
                 $imageCache->generate($product);
 
-                $output->write(".");
+                $this->productRepository->cleanCache();
```
Comment (okorshenko) [inline app/code/Magento/Catalog/Console/Command/ImagesResizeCommand.php:97]: this method is out of RepositoryInterface, so we can not call it directly. As a workaround, we can check if method_exists and call it. In this case the fix will be BC

**Review by okorshenko — CHANGES_REQUESTED** [review]

File: app/code/Magento/Catalog/Api/ProductRepositoryInterface.php (hunk 1 above)
Comment (Koc) [reply app/code/Magento/Catalog/Api/ProductRepositoryInterface.php:81]: I will try backport another implementation from develop brahcn.

**Comment by okorshenko** [comment]
Closing this PR due to inactivity. provided changes do not follow Backward compatibility policy. 
Thank you @Koc for your effort

**Pull request closed without merging**